doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -373,7 +375,16 @@ the propositional implication and function types.
 Applications
 ------------
 
-:n:`@term__fun @term` denotes applying the function :n:`@term__fun` to :token:`term`.
+:n:`@term__fun @term` denotes applying the :term:`function`
+:n:`@term__fun` to :token:`term`.
+
+.. glossary::
+
+   Function
+     Any Coq object whose type is a :ref:`product <product>`.
+
+     In particular, a non-dependent function has a non-dependent
+     product type :n:`@type -> @type`.
 
```

Comment:
This shows up in the general index like this:

![image](https://user-images.githubusercontent.com/1253341/79610034-a0e03580-80ac-11ea-887b-693c8f50bd6d.png)

This is actually 2 entries; the first is the glossary term and the second is for the `Function` command.  A bit confusing.

Also, I think we should have the glossary as a separate index.  That might include both glossary terms and concepts (concepts can be phrases).

Reply:
I personally cannot stand the general index. But what you see is a general issue. Indexed names are not repeated. This is pretty bad IMHO.

Reply:
Yes, the general index is not very useful, but there is no index for the glossary yet, so I looked there.